### PR TITLE
add paginator to policy list

### PR DESCRIPTION
**Pull request by RussellChamp**
this is the FE part to match the BE work I did today

### CODE REVIEW

File: src/components/Paginator.svelte [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+export let thingName = 'items'
+export let limitOptions = [10, 20, 50, 100]
+export let limitSelection = ''
+
+$: limitSelectOptions = limitOptions.map((o) => ({ id: 'limit-' + o, name: o } as SelectOption))
+
+const onClickFirst = () => {
+  dispatch('paginate', { page: 1, limit: pageData.per_page })
+}
+const onClickLast = () => {
```
Comment (Schparky) [inline src/components/Paginator.svelte:20]: Did we stop adding blank lines between functions?

File: src/components/Paginator.svelte [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  if (pageData.page < pageData.total_pages) {
+    dispatch('paginate', { page: pageData.page + 1, limit: pageData.per_page })
+  }
+}
+const onSetLimitDefault = () => {
+  if (!limitSelection && limitSelectOptions?.length > 0) limitSelection = limitSelectOptions[0].id
+}
+const onLimitChange = (event: CustomEvent<SelectOption>) => {
+  if (event.detail.name) {
+    // re-jigger the page number such that the current slice of data (at least the first item) is in the newly selected page
```
Comment (Schparky) [inline src/components/Paginator.svelte:38]: nit: if I understand correctly, the comment applies to the `let newPage` line, in which case the comment should be bumped down one line

File: src/data/policies.ts [hunk 3]
```diff
@@ -129,28 +148,26 @@ export const getNameOfPolicy = (policy: Policy): string => {
...
 }
 
-export async function searchPoliciesFor(searchText: string, limit = 50): Promise<Policy[]> {
-  const queryString = qs.stringify({ search: searchText, limit })
-  const response = await GET<{ data: Policy[]; meta: any }>(`policies?${queryString}`)
-  return response.data
+export async function searchPoliciesFor(searchText: string, page = 1, limit = 20): Promise<GetPoliciesResponseBody> {
+  const queryString = qs.stringify({ search: searchText, page, limit })
+  const response = await GET<GetPoliciesResponseBody>(`policies?${queryString}`)
+  return response
```
Comment (Schparky) [inline src/data/policies.ts:170]: Not related to this change, but I wonder if this should also call `updatePolicyStore`

File: src/pages/admin/policies.svelte [hunk 4]
```diff
@@ -1,25 +1,52 @@
...
+import { Paginator, SearchForm } from 'components'
 import { getNameOfPolicy, Policy, PolicyType, searchPoliciesFor } from 'data/policies'
 import type { PolicyMember } from 'data/policy-members'
 import { urlQuery } from 'data/query-string'
 import { formatPageTitle } from 'helpers/pageTitle'
 import { ADMIN_POLICIES, adminPolicySearch, policyDetails } from 'helpers/routes'
 import { goto, metatags } from '@roxi/routify'
 import { Datatable, Page } from '@silintl/ui-components'
+import qs from 'qs'
+import { onMount } from 'svelte'
```
Comment (Schparky) [inline src/pages/admin/policies.svelte:11]: sort imports

File: src/components/Paginator.svelte [hunk 5]
```diff
@@ -0,0 +1,95 @@
+<script lang="ts">
+import { IconButton, Select } from '@silintl/ui-components'
+import type { PaginatedData } from 'data/types/PaginatedData'
+import { createEventDispatcher } from 'svelte'
+
+type SelectOption = { id: string; name: string | number }
```
Comment (Schparky) [inline src/components/Paginator.svelte:6]: can `name` be just `number`?

**Review by Schparky — APPROVED** [review]
good work!

File: src/components/Paginator.svelte [hunk 6]
```diff
@@ -0,0 +1,95 @@
+<script lang="ts">
+import { IconButton, Select } from '@silintl/ui-components'
+import type { PaginatedData } from 'data/types/PaginatedData'
```
Comment (hobbitronics) [inline src/components/Paginator.svelte:3]: sort imports

File: src/pages/admin/policies.svelte [hunk 7]
```diff
@@ -1,25 +1,52 @@
...
+import type { PaginatedData } from 'data/types/PaginatedData'
 
 let matchingPolicies: Policy[] = []
+let pageData: PaginatedData = {} as PaginatedData
```
Comment (hobbitronics) [inline src/pages/admin/policies.svelte:15]: I'm curious why you don't just do `let pageData = {} as PaginatedData`

File: src/pages/admin/policies.svelte [hunk 8]
```diff
@@ -1,25 +1,52 @@
...
 let searchText = ''
 
 metatags.title = formatPageTitle('Policies')
 
+onMount(() => search())
+
 $: searchText = $urlQuery.search || ''
-$: searchPoliciesFor(searchText).then((result) => (matchingPolicies = result))
+$: pageData.page = Number($urlQuery.page) || 1
+$: pageData.per_page = Number($urlQuery.limit) || 20
```
Comment (hobbitronics) [inline src/pages/admin/policies.svelte:24]: is `$params.page` the same thing as `$urlQuery.page`?

Comment (RussellChamp) [reply src/pages/admin/policies.svelte:24]: not sure. i was following the same pattern that searchText already was using

**Pull request merged**